### PR TITLE
[braintree] fix search api to be a stream instead of a promise

### PR DESCRIPTION
**Pull request by sharno**
Reference to stream changes of braintree search api:
https://developers.braintreepayments.com/reference/request/subscription/search/node

### CODE REVIEW

File: types/braintree/index.d.ts [hunk 1]
```diff
@@ -16,11 +20,11 @@ declare namespace braintree {
         Development = 'Development',
         Production = 'Production',
         Qa = 'Qa',
-        Sandbox = 'Sandbox'
+        Sandbox = 'Sandbox',
     }
 
     export interface GatewayConfig {
-        environment: any;
+        environment: Environment;
```
Comment (smrubin) [inline types/braintree/index.d.ts]: 👍

File: types/braintree/braintree-tests.ts [hunk 2]
```diff
@@ -23,7 +34,8 @@ const gateway: BraintreeGateway = new braintree.BraintreeGateway({
         customerId: '123456',
         streetAddress: '222 Oak Street',
     };
-    const response = await gateway.address.create(addressRequest);
+    const response = await gateway.address.create(addressRequest).catch(console.error);
+    if (!response) return;
```
Comment (smrubin) [inline types/braintree/braintree-tests.ts]: Yeah, I suppose this is a good idea for other contributors' envs where they might not have a gateway setup.

File: types/braintree/index.d.ts [hunk 3]
```diff
@@ -94,14 +102,14 @@ declare namespace braintree {
     }
 
     interface CreditCardVerificationGateway {
-        search(searchFn: any): Promise<CreditCardVerification[]>;
+        search(searchFn: any): stream.Readable;
```
Comment (smrubin) [inline types/braintree/index.d.ts]: @sharno Thanks for fixing search and adding in streams in. Do you know if there is there a way to add typings to the stream objects (i.e. `CreditCardVerification` here)? That could be very nice to have.

**Review by smrubin — APPROVED** [review]
Looks good @sharno. Thanks for your work on `search` and the linting as well.

Had one comment about adding types to the stream objects, but that could be a follow-up to this PR, if it's possible and you're interested in taking on.

File: types/braintree/index.d.ts (hunk 3 above)
Comment (sharno) [reply types/braintree/index.d.ts]: I tried to look for something like this but I couldn't find any. It's much helpful of course if we could type the stream but it doesn't look the be the case in `@types/node`.

A solution might be looking into if it's possible to make `@types/node`'s `stream.Readable` take a type param.

A custom stream type for braintree is possible but wouldn't work well with the other stream types from `@types/node`

File: types/braintree/index.d.ts (hunk 1 above)
Comment (sharno) [reply types/braintree/index.d.ts]: Thanks a lot for reviewing and approving this PR

**Pull request merged**